File: src/DependencyGraph.cs
namespace graph{
    public class DependencyGraph{
        List<Node> nodes { get; }

        public DependencyGraph(){
            this.nodes = new List<Node>();
        }

        //add new node to graph
        public Node addNode(string codeblock){
            var newNode = new Node(codeblock);
            nodes.Add(newNode);
            return newNode;
        }

        //add neighbor to node
        public void addNeighbor(Node x, Node y, string relationship){
            x.getNeighbors().Add(new Tuple<Node,string>(y, relationship));
        }

        public List<Node> getNodes(){
            return nodes;
        }

    
    }   
}

File: src/DependencyGraph.js
class DependencyGraph{
    constructor(){
        //keep track of adjacent nodes
        this.adjList = new Map();
        //keep track of relationships between nodes
        this.edgeRelations = new Map();
    }

    //add node to graph
    addNode(n){
        this.adjList.set(n,[]);
    }

    //dependency graph is undirected
    addEdge(n, nn, relation1, relation2){
        this.AdjList.get(n).push(nn);
        this.AdjList.get(nn).push(n);
        this.edgeRelations.set(n, [nn, relation1]);
        this.edgeRelations.set(nn, [n, relation2]);
    }
}